### ISSUE TITLE
Migrations fail on clean installs when using MySQL-like and PHP8

### ISSUE THREAD

**Issue by cesare190** [shlinkio/shlink#1047]
Good Morning, I apologize for this message, it is the first report I make on Github, not knowing all the best practices I hope to do something pleasant by reporting a possible bug that I am experiencing on Shlink in its latest version as of today with Ubuntu 20.04, Apache and PHP 8 updated to its latest version as of today.

I have had some problems using Shlink since PHP 8 is available, so for simplicity I decided to do it all over again except for the VHOST configuration which remained the same (I only adjusted the Php Fastcgi socket).

In the final stage of the installation, once I ran the ./install -vvv command I had the following final output:

```
Initializing database ... [Running" /usr/bin/php8.0 bin / cli db: create "] RUN '/usr/bin/php8.0' 'bin / cli' 'db: create'
  OUT Creating database tables ...
  OUT
  OUT [OK] Database properly created!
  OUT
  OUT
  RES Command ran successfully
 Success!
Updating database ... [Running "/usr/bin/php8.0 bin / cli db: migrate"] RUN '/usr/bin/php8.0' 'bin / cli' 'db: migrate'
  OUT Migrating database ...
  OUT
  ERR
  ERR In Process.php line 267:
  ERR
  ERR The command "'/usr/bin/php8.0' 'vendor / doctrine / migrations / bin / doctrine-mig
  ERR rations.php '' migrations: migrate '' --no-interaction '"failed.
  ERR
  ERR Exit Code: 1 (General error)
  ERR
  ERR Working directory: / var / www / shorturl
  ERR
  ERR Output:
  ERR ================
  ERR
  ERR
  ERR Error Output:
  ERR ================
  ERR
  ERR In Connection.php line 1761:
  ERR
  ERR There is no active transaction
  ERR
  ERR
  ERR migrations: migrate [--write-sql [WRITE-SQL]] [--dry-run] [--query-time] [-
  ERR allow-no-migration] [--all-or-nothing [ALL-OR-NOTHING]] [--configuration CO
  ERR NFIGURATION] [--em EM] [--conn CONN] [-] [<version>]
  ERR
  ERR
  ERR db: migrate
  ERR
  ERR
  RES 1 Command did not run successfully
```

I tried to check if the installation was "usable" anyway. I managed to generate an api key and create a shortlink but that link doesn't redirect properly.

The steps in summary were

1) run the command ./cli api-key: generate
2) copy the API key obtained
3) go to the webclient and access the personal area to be authenticated
4) create a link to google.it without customizations
5) copy the shorturl obtained and then visit it
6) see the error "An unexpected error occurred"

The error log, managed by Apache through vhost, is empty.

If I try to generate a shorturl from the ubuntu terminal via ./cli short-url: generate and providing https://google.it (example link) this error appears:

`PHP Fatal error:  Uncaught Error: Undefined constant "IDNA_DEFAULT" in /var/www/shorturl/vendor/guzzlehttp/guzzle/src/Client.php:217
Stack trace:
#0 /var/www/shorturl/vendor/guzzlehttp/guzzle/src/Client.php(160): GuzzleHttp\Client->buildUri()
#1 /var/www/shorturl/vendor/guzzlehttp/guzzle/src/Client.php(187): GuzzleHttp\Client->requestAsync()
#2 /var/www/shorturl/module/Core/src/Util/UrlValidator.php(73): GuzzleHttp\Client->request()
#3 /var/www/shorturl/module/Core/src/Util/UrlValidator.php(56): Shlinkio\Shlink\Core\Util\UrlValidator->validateUrlAndGetResponse()
#4 /var/www/shorturl/module/Core/src/ShortUrl/Helper/ShortUrlTitleResolutionHelper.php(25): Shlinkio\Shlink\Core\Util\UrlValidator->validateUrlWithTitle()
#5 /var/www/shorturl/module/Core/src/Service/UrlShortener.php(49): Shlinkio\Shlink\Core\ShortUrl\Helper\ShortUrlTitleResolutionHelper->processTitleAndValidateUrl()
#6 /var/www/shorturl/module/CLI/src/Command/ShortUrl/GenerateShortUrlCommand.php(165): Shlinkio\Shlink\Core\Service\UrlShortener->shorten()
#7 /var/www/shorturl/vendor/symfony/console/Command/Command.php(256): Shlinkio\Shlink\CLI\Command\ShortUrl\GenerateShortUrlCommand->execute()
#8 /var/www/shorturl/vendor/symfony/console/Application.php(971): Symfony\Component\Console\Command\Command->run()
#9 /var/www/shorturl/vendor/symfony/console/Application.php(290): Symfony\Component\Console\Application->doRunCommand()
#10 /var/www/shorturl/vendor/symfony/console/Application.php(166): Symfony\Component\Console\Application->doRun()
#11 /var/www/shorturl/config/run.php(14): Symfony\Component\Console\Application->run()
#12 /var/www/shorturl/bin/cli(7): {closure}()
#13 {main}
  thrown in /var/www/shorturl/vendor/guzzlehttp/guzzle/src/Client.php on line 217`

I know I have not followed the guidelines, I apologize for this lack

Have a nice day,
Massimiliano

**Comment by patrickhilker** [comment]
Got the same error. I assume [this Issue in the Doctrine project](https://github.com/doctrine/migrations/issues/1104) belongs to the error.

**Comment by acelaya** [comment]
I se two separate issues here.

* Migrations error: it's most probably what @patrickhilker is linking, but if you run the install script with `-vvv`, well know for sure.
  There's no easy workaround for this. I applied the recommended one there to new migrations, but not existing ones. I'll try the latter and see if it works.
* Missing IDNA_DEFAULT constant: this one it's because you need to install the intl extension.

**Comment by cesare190** [comment]
Ok i

> I se two separate issues here.
> 
> * Migrations error: it's most probably what @patrickhilker is linking, but if you run the install script with `-vvv`, well know for sure.
>   There's no easy workaround for this. I applied the recommended one there to new migrations, but not existing ones. I'll try the latter and see if it works.
> * Missing IDNA_DEFAULT constant: this one is because you need to install the intl extension.

I can check for the second point. 

yes, there are two points but I was convinced that they were both to be attributed to the first

**Comment by patrickhilker** [comment]
> I applied the recommended one there to new migrations, but not existing ones. I'll try the latter and see if it works.

I assume you mean you added the method `isTransactional(): bool` returning `false` to the migrations. In a bigger (closed source) private project of mine I also did this, even to older migrations. Works like a charm for me.

**Comment by acelaya** [comment]
Actually, on a fresh Shlink installation, it's not that important to run the migration. They are run just to mark all of them as executed, but everything should work anyway.

It's more important when updating Shlink though. In that case, failing to run them will usually result in Shlink not working afterwards, to some extent.

**Comment by acelaya** [comment]
> I assume you mean you added the method isTransactional(): bool returning false to the migrations. In a bigger (closed source) private project of mine I also did this, even to older migrations. Works like a charm for me (only tested with PHP 8 as I do not have to maintain older versions).

Yes exactly. I added it only to a few newer migrations added after this issue was discovered. I was a bit worried some nasty side effect could be introduced by editing older ones, but I'll give it a try as this is affecting new installations.

Thanks for the hint.

**Comment by cesare190** [comment]
> I se two separate issues here.
> 
> * Migrations error: it's most probably what @patrickhilker is linking, but if you run the install script with `-vvv`, well know for sure.
>   There's no easy workaround for this. I applied the recommended one there to new migrations, but not existing ones. I'll try the latter and see if it works.
> * Missing IDNA_DEFAULT constant: this one is because you need to install the intl extension.

with the intl the redirect seems to go. I keep having problems with the clean install

**Comment by cesare190** [comment]
@acelaya I used MariaDB not MySQL

**Comment by acelaya** [comment]
Hey @cesare190, @patrickhilker, I have just released v2.6.2 which fixes the migrations error when using MySQL/Mariadb and PHP8.
